### PR TITLE
Tables exception and error classes changes.

### PR DESCRIPTION
**Pull request by vcolin7**
- Renamed TableServiceErrorException to TableServiceException.
- Made `TableErrorCode` final and added a lot more error codes.
- Removed `languageCode` from `TableServiceError`.

Unrelated to the title but also a change made here:
- Moved the code snippets classes to the correct package (`com.azure.data.tables`).

### CODE REVIEW

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceAsyncClient.java [hunk 1]
```diff
@@ -223,7 +223,7 @@ public TableAsyncClient getTableClient(String tableName) {
      * @param tableName The name of the table to delete.
      * @return A reactive result containing the HTTP response.
      * @throws IllegalArgumentException If {@code tableName} is {@code null} or empty.
-     * @throws TableServiceErrorException If no table with the provided name exists within the service.
+     * @throws TableServiceException If no table with the provided name exists within the service.
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceAsyncClient.java:226]: I'm guessing the 404-swallowing will be in a separate PR?

Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceAsyncClient.java:226]: Correct

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableErrorCode.java [hunk 2]
```diff
@@ -8,43 +8,277 @@
...
+public final class TableErrorCode extends ExpandableStringEnum<TableErrorCode> {
+    /**
+     * Static value {@code InvalidTableName}.
+     */
+    public static final TableErrorCode INVALID_TABLE_NAME = fromString("InvalidTableName");
+
+    /**
+     * Static value {@code InvalidPkOrRkName}.
+     */
+    public static final TableErrorCode INVALID_PK_OR_RK_NAME = fromString("InvalidPkOrRkName");
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableErrorCode.java:20]: Why do these two not have the "Value" suffix?

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceAsyncClient.java (hunk 1 above)
Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceAsyncClient.java:226]: Although I could add it on this one if you feel it makes sense.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableErrorCode.java [hunk 3]
```diff
@@ -8,43 +8,277 @@
...
+    /**
+     * Static value {@code Forbidden}.
+     */
+    public static final TableErrorCode FORBIDDEN = fromString("ForbiddenValue");
+
+    /**
+     * Static value {@code AuthorizationPermissionMismatch}.
+     */
+    public static final TableErrorCode AUTHORIZATION_PERMISSION_MISMATCH =
+        fromString("AuthorizationPermissionMismatch");
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableErrorCode.java:274]: This one also doesn't have "Value"

File: sdk/tables/azure-data-tables/src/samples/java/com/azure/data/tables/TableServiceClientCodeSnippets.java [hunk 4]
```diff
@@ -27,10 +27,10 @@ public void createTable() {
             .buildClient();
         try {
             tableServiceClient.createTable("OfficeSupplies");
-        } catch (TableStorageException e) {
```
Comment (annatisch) [inline sdk/tables/azure-data-tables/src/samples/java/com/azure/data/tables/TableServiceClientCodeSnippets.java:30]: Interesting that this had `TableStorageException` - Or was this sample previously incorrect?

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableErrorCode.java (hunk 3 above)
Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableErrorCode.java:274]: None should have the value suffix, it was a mistake when copying the new entries from .NET

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableErrorCode.java (hunk 2 above)
Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableErrorCode.java:20]: None should have the value suffix, it was a mistake when copying the new entries from .NET

File: sdk/tables/azure-data-tables/src/samples/java/com/azure/data/tables/TableServiceClientCodeSnippets.java (hunk 4 above)
Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/samples/java/com/azure/data/tables/TableServiceClientCodeSnippets.java:30]: It was incorrect before. From what I can tell, `TableStorageException` was a type added at some point to expose error information in place of the internal exception type, but was never really used broadly, which is why I introduced `TableServiceException`.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceAsyncClient.java (hunk 1 above)
Comment (annatisch) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceAsyncClient.java:226]: Nah - separate is good.

File: sdk/tables/azure-data-tables/src/samples/java/com/azure/data/tables/TableServiceClientCodeSnippets.java (hunk 4 above)
Comment (annatisch) [reply sdk/tables/azure-data-tables/src/samples/java/com/azure/data/tables/TableServiceClientCodeSnippets.java:30]: Seems like there were previously two `TableStorageException` and `TableServiceErrorException`, that are now resolved to the single `TableServiceException`?

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableErrorCode.java (hunk 3 above)
Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableErrorCode.java]: Good catch!

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableErrorCode.java (hunk 2 above)
Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableErrorCode.java:20]: Good catch!

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableErrorCode.java (hunk 3 above)
Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableErrorCode.java]: Good catch!

File: sdk/tables/azure-data-tables/src/samples/java/com/azure/data/tables/TableServiceClientCodeSnippets.java (hunk 4 above)
Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/samples/java/com/azure/data/tables/TableServiceClientCodeSnippets.java:30]: Yes, every error that comes from the service in the form of the generated `TableServiceErrorException` is now mapped to the public `TableServiceException`.

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableErrorCode.java [hunk 5]
```diff
@@ -8,7 +8,65 @@
 /**
  * Defines values for error codes returned from the Tables service.
  */
-public class TableErrorCode extends ExpandableStringEnum<TableErrorCode> {
+public final class TableErrorCode extends ExpandableStringEnum<TableErrorCode> {
+    /**
+     * Static value {@code InvalidTableName}.
+     */
+    public static final TableErrorCode INVALID_TABLE_NAME = fromString("InvalidTableName");
```
Comment (srnagar) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableErrorCode.java:15]: Where are these error codes documented? Can we add a link to it in the JavaDoc? Also, is this something that we can include in the swagger instead of manually enumerating each error code coming from the service?

Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableErrorCode.java:15]: Error codes can be found here:

- [Common Storage error codes](https://docs.microsoft.com/en-us/rest/api/storageservices/common-rest-api-error-codes)
- [Tables Specific codes](https://docs.microsoft.com/en-us/rest/api/storageservices/table-service-error-codes)

Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableErrorCode.java:15]: Would the link be documented on the Exception and Error classes or somewhere else?

Comment (vcolin7) [reply sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableErrorCode.java:15]: I think we can ask the service team to include them in the swagger, what do you think @annatisch?

File: sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableErrorCode.java [hunk 6]
```diff
@@ -6,9 +6,59 @@
 import com.azure.core.util.ExpandableStringEnum;
 
 /**
- * Defines values for error codes returned from the Tables service.
+ * Defines values for error codes returned from the Tables service. For detailed information about what each error
+ * code means and their accompanying status codes, visit the following page:
+ * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/table-service-error-codes">Table service error codes</a>
```
Comment (srnagar) [inline sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/models/TableErrorCode.java:11]: Remove `en-us` locale from the link.

**Review by srnagar — APPROVED** [review]
